src/main/java/com/clouway/push/server/PushServiceImpl.java

```diff
@@ -36,16 +35,13 @@ public void pushEvent(PushEvent event) {
   @Override
   public void pushEvent(PushEvent event, String correlationId) {
 
-    // transforming the eventType
-    if (!Strings.isNullOrEmpty(correlationId)) {
-      event.getAssociatedType().setCorrelationId(correlationId);
-    }
+    String key = event.getKey() + correlationId;
 
     String message = eventSerializer.serialize(event);
```

Comment:
that was my first solution, but there is a method pushService.push(event, correlationId); which is outside of the object, so I will create a wrapper object that will contains the event and the correaltionId, so they will be transported separetly to the client.